### PR TITLE
feat(mobile): replace positions accordion with bottom sheet and update tokens menu

### PR DESCRIPTION
**Pull request by compojoom**
## Summary

- **Positions tab**: Replaced the accordion expand/collapse pattern with a flat protocol list. Tapping a protocol now opens a bottom sheet showing protocol details (icon, name, percentage badge, fiat total, 24h change) and grouped positions
- **Tokens tab**: Renamed "Hide suspicious tokens" to "Show all tokens" (inverted toggle), added new "Hide small balances" toggle filtering tokens with fiat value < $0.01 (matching web app behavior)
- Added `hideDust` setting to Redux store with persist migration support

### Positions Bottom Sheet
- Refactored `ProtocolSection.tsx` — removed accordion, renders flat row with chevron, navigates to bottom sheet on press
- Created `ProtocolDetailSheet` (presentational + container) with protocol header and grouped positions
- Created `protocol-detail-sheet.tsx` route registered as transparent modal in `_layout.tsx`
- Added `calculateProtocolFiatChange()` utility aggregating position-level 24h changes

### Tokens Menu
- Updated `ManageTokensSheet.tsx` — renamed toggle label, added second toggle for dust filtering
- Updated `ManageTokensSheet.container.tsx` — wired `hideDust` state from Redux
- Updated `Tokens.container.tsx` — client-side dust filtering with `DUST_THRESHOLD` constant
- Extended `NoFunds` component with optional title/description for filtered-out empty state

### Store
- Added `hideDust: boolean` field, `setHideDust` reducer, and `selectHideDust` selector to `settingsSlice.ts`
- Added `DUST_THRESHOLD = 0.01` constant to `config/constants.ts`

## Test plan

- [x] `settingsSlice.test.ts` — hideDust default, toggle, persist migration
- [x] `ManageTokensSheet.test.tsx` — both toggles render and fire callbacks
- [x] `ProtocolSection.test.tsx` — flat row rendering, navigation on press, no inline positions
- [x] `ProtocolDetailSheet/utils.test.ts` — fiat change aggregation edge cases
- [x] `Positions.container.test.tsx` — updated for no inline position rendering
- [x] All 227 test suites pass (1922 tests)
- [x] Type-check clean
- [x] Lint clean

## Screenshots

https://github.com/user-attachments/assets/ed13d8c0-962c-4432-88ad-31727eeef56d

https://github.com/user-attachments/assets/6f7ab8ef-b72b-46f3-b37b-9ab781dc2bb0

### CODE REVIEW

File: apps/mobile/src/features/Assets/components/Tokens/Tokens.container.tsx [hunk 1]
```diff
@@ -43,6 +44,20 @@ export function TokensContainer() {
     },
   )
 
+  const shouldFilterDust = hideDust && hasDefaultTokenlist
+
+  const visibleItems = useMemo(() => {
+    if (!data?.items) {
+      return undefined
```
Comment (clovisdasilvaneto) [inline apps/mobile/src/features/Assets/components/Tokens/Tokens.container.tsx]: why not just `return`?

File: apps/mobile/src/features/Assets/components/Positions/ProtocolDetailSheet/ProtocolDetailSheet.tsx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+import type { Protocol } from '@safe-global/store/gateway/AUTO_GENERATED/positions'
+import { calculateProtocolFiatChange } from './utils'
+
+interface ProtocolDetailSheetHeaderProps {
+  protocol: Protocol
+  percentageRatio: number
+  currency: string
+}
+
+export const ProtocolDetailSheetHeader = ({ protocol, percentageRatio, currency }: ProtocolDetailSheetHeaderProps) => {
```
Comment (clovisdasilvaneto) [inline apps/mobile/src/features/Assets/components/Positions/ProtocolDetailSheet/ProtocolDetailSheet.tsx]: could we move this component to a separated file?
